assets/js/modules/analytics-4/components/common/MeasurementSelect.js

```diff
@@ -0,0 +1,200 @@
+import { ProgressBar } from 'googlesitekit-components';
+import { Select, Option } from '../../../../material-components';
+import {
+	MODULES_ANALYTICS_4,
+	PROPERTY_CREATE,
+	WEBDATASTREAM_CREATE,
+} from '../../datastore/constants';
+import { MODULES_ANALYTICS } from '../../../analytics/datastore/constants';
+import { isValidAccountID } from '../../../analytics/util';
+import { isValidWebDataStreamSelection } from '../../utils/validation';
+import { trackEvent } from '../../../../util';
+import useViewContext from '../../../../hooks/useViewContext';
+const { useSelect, useDispatch } = Data;
+
+export default function MeasurementSelect( props ) {
+	const { hasModuleAccess, className } = props;
+
+	// Analytics accounts need to be loaded in order to load the properties,
+	// otherwise this component will stay in a loading state forever.
+	// eslint-disable-next-line no-unused-vars
+	useSelect( ( select ) => select( MODULES_ANALYTICS ).getAccounts() );
+
+	// TODO: Update this select hook to pull accountID from the modules/analytics-4
+	// datastore when GA4 module becomes separated from the Analytics one.
+	const accountID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).getAccountID()
+	);
+
+	const { propertyID, webDataStreamID, measurementID } = useSelect(
+		( select ) => select( MODULES_ANALYTICS_4 ).getSettings() || {}
+	);
+
+	const isRealPropertyID = !! propertyID && propertyID !== PROPERTY_CREATE;
+	const webDataStreams = useSelect( ( select ) =>
+		isRealPropertyID
+			? select(
+					MODULES_ANALYTICS_4
+			  ).getMatchingWebDataStreamsByPropertyID( propertyID )
+			: []
+	);
+
+	const isLoading = useSelect(
+		( select ) =>
+			! select( MODULES_ANALYTICS ).hasFinishedResolution(
+				'getAccounts'
+			) ||
+			! select( MODULES_ANALYTICS_4 ).hasFinishedResolution(
+				'getProperties',
+				[ accountID ]
+			) ||
+			! select( MODULES_ANALYTICS_4 ).hasFinishedResolution(
+				'getWebDataStreams',
+				[ propertyID ]
+			) ||
+			select( MODULES_ANALYTICS ).hasFinishedSelectingAccount() === false
+	);
+
+	const viewContext = useViewContext();
+	const { setWebDataStreamID, updateSettingsForMeasurementID } =
+		useDispatch( MODULES_ANALYTICS_4 );
+	const onWebDataStreamChange = useCallback(
+		( index, { dataset } ) => {
+			const newID = dataset.value;
+			if ( webDataStreamID === newID ) {
+				return;
+			}
+
+			setWebDataStreamID( newID );
+			updateSettingsForMeasurementID(
+				webDataStreams.find( ( { _id } ) => _id === newID )
+					?.webStreamData?.measurementId || '' // eslint-disable-line sitekit/acronym-case
+			);
+
+			trackEvent(
+				`${ viewContext }_analytics`,
+				newID === WEBDATASTREAM_CREATE
+					? 'change_webdatastream_new'
+					: 'change_webdatastream',
+				'ga4'
+			);
+		},
+		[
+			webDataStreams,
+			webDataStreamID,
+			setWebDataStreamID,
+			updateSettingsForMeasurementID,
+			viewContext,
+		]
+	);
+
+	if ( ! isValidAccountID( accountID ) || ! isRealPropertyID ) {
+		return null;
```

Comment:
This component should also display when a new property is chosen, similar to UA
<img width="470" alt="image" src="https://user-images.githubusercontent.com/1621608/224379909-2d8bb945-638d-4f00-be40-57f193d2c9ff.png">

Reply:
Yep. Updated.